pkg/process/checks/process_fallback_test.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !linux
+
+package checks
+
+import (
+	"testing"
+	"time"
+
+	wmimpl "github.com/DataDog/datadog-agent/comp/core/workloadmeta/impl"
+	probemocks "github.com/DataDog/datadog-agent/pkg/process/procutil/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestProcessByPID ensures the usage of the probe when wlm collection is ON/OFF
+func TestProcessByPID(t *testing.T) {
+	for _, tc := range []struct {
+		description      string
+		useWLMCollection bool
+		collectStats     bool
+	}{
+		{
+			description:      "wlm collection ENABLED, with stats ENABLED",
+			useWLMCollection: true,
+			collectStats:     true,
+		},
+		{
+			description:      "wlm collection ENABLED, with stats DISABLED",
+			useWLMCollection: true,
+			collectStats:     false,
+		},
+		{
+			description:      "wlm collection DISABLED, with stats ENABLED",
+			useWLMCollection: false,
+			collectStats:     true,
+		},
+		{
+			description:      "wlm collection DISABLED, with stats DISABLED",
+			useWLMCollection: false,
+			collectStats:     false,
+		},
+	} {
```

Comment:
I think this test can be simplified to remove cases based on `collectStats` it doesn't change anything about the test. It's just a value passed into the function.